storage-s3-dynamodb/src/main/java/io/delta/storage/utils/ReflectionUtils.java

```diff
@@ -0,0 +1,37 @@
+package io.delta.storage.utils;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import org.apache.hadoop.conf.Configuration;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+
+public class ReflectionUtils {
+
+
+    private static boolean readsCredsFromHadoopConf(Class<?> awsCredentialsProviderClass) {
+        return Arrays.stream(awsCredentialsProviderClass.getConstructors())
+                .anyMatch(constructor -> constructor.getParameterCount() == 1 &&
+                        Arrays.equals(constructor.getParameterTypes(), new Class[]{Configuration.class}));
+    }
+
+    public static AWSCredentialsProvider createAwsCredentialsProvider(
+            String credentialsProviderName,
+            Configuration initHadoopConf) throws ClassNotFoundException,
+            InstantiationException,
+            IllegalAccessException,
+            InvocationTargetException,
+            NoSuchMethodException {
+        Class<?> awsCredentialsProviderClass = Class.forName(credentialsProviderName);
+        final AWSCredentialsProvider awsCredentialsProvider;
+        if (readsCredsFromHadoopConf(awsCredentialsProviderClass))
+            awsCredentialsProvider = (AWSCredentialsProvider) awsCredentialsProviderClass
```

Comment:
nit: just return the created cred provider rather than storing in a var and returning it. same for the else loop.

Reply:
done.